packages/flame/lib/src/components/component.dart

```diff
@@ -331,26 +331,35 @@ class Component {
 
   /// Recursively enumerates all nested [children].
   ///
-  /// The search is depth-first, hence descendants are in postorder. In other
-  /// words, it explores the first child completely before visiting the next
-  /// sibling.
-  ///
-  /// In order to filter descendants it is usually convenient to use [Iterable]
-  /// lazy methods, such as [Iterable.where].
-  Iterable<Component> descendants({bool includeSelf = false}) sync* {
-    if (includeSelf) {
+  /// The search is depth-first in preorder. In other words, it explores the
+  /// first child completely before visiting the next sibling, and the root
+  /// component is visited before its children.
+  ///
+  /// This ordering of descendants is considered standard in Flame: it is the
+  /// same order in which the components will normally be updated and rendered
+  /// on every game cycle. The optional parameter [reversed] allows iterating
+  /// through the same set of descendants in reverse order.
+  ///
+  /// The [Iterable] produced by this method is "lazy", which means it will only
+  /// traverse the component tree when required. This allows efficient chaining
+  /// of various iterable methods, such as filtering, early stopping, folding,
+  /// and so on -- see the documentation of the [Iterable] class for details.
+  Iterable<Component> descendants({
+    bool includeSelf = false,
+    bool reversed = false,
```

Comment:
Maybe we should have this flag on ancestors too?

Reply:
I don't think there is a way to implement this for ancestors more efficiently than `ancestors().toList().reversed`.

Reply:
True, since we don't have access to the root that it is attached to. Let's skip that then.